CapuaEnvironment/MemoryArray/MemoryArray.py

```diff
@@ -76,12 +75,24 @@ def extractMemory(self, address, length=1):
 
         return memorySlice
 
+    def writeMemory(self, address, values, length):
+        """
+        This method will overwrite values from a given address
+        :param address: int, the address where the overwrite is to happen
+        :param values: int list, a list containing values to be writen in memory
+        :param length: int, length of the write operation
+        :return: none
+        """
+        baseIndex = self._computeArrayIndexFromAddress(address)
+        for i in range(0, length):
+            self._memoryCellArray[baseIndex + i] = values[i]
```

Comment:
A few style changes here:
- `writeMemory` is taking in `length` and `values`, `values` being a `List` already provides a length using `len`. Allowing length to be specified is redundant and can allow user error. 
- Manual indexing isn't necessary.

I'd suggest the following changes  
```python 3
    def writeMemory(self, address, values):
        """
        This method will overwrite values from a given address
        :param address: int, the address where the overwrite is to happen
        :param values: int list, a list containing values to be writen in memory
        :return: none
        """        
        base = self._computeArrayIndexFromAddress(address) 
        self._memoryCellArray[base:base+len(values)] = values
```
One thing to note is what happens if we go over the range and how it should be handled.

I double checked what `length` is used for in the rest of the function and it's still necessary for other calls( without adding validation to verify inputs such as (0b0010) take less than 4 bytes.)

Note about style: seems like other functions will use simplifications like this where possible. I.E. `extractMemory`

Reply:
Very good call! Was not aware a slice could be replaced in a single go.

Reply:
As for simplification and removal of the length param from the various functions, I think this should be part of a different PR in order to keep the code style uniform.